### DISCUSSION TITLE
Asset ID is linked to the path instead of being a UID

### DISCUSSION THREAD

**Discussion by kevinmeijer97** [statamic/cms#6479] [Troubleshooting]
When an asset is set on a collection and the asset is moved to a different folder, the asset doesn't work anymore on the collection because the ID is changed according to the path it is saved on.
That means if we have an image in /folder-a/image.png and we move it to folder-b/image.png, the image does not work on all locations where folder-a/image.png is used because the ID is also changed relative to the path.

Is this something that was done on purpose, and is it possible to create a fix to where either:

- All places where folder-a/image.png are being used will be changed to the new ID.
- Created an actual UID when uploading an image, so that when the image is moved the ID doesn't change.

I have also seen [this PR](https://github.com/statamic/cms/pull/4832), where it seems like an option is being added to replace assets. Is it true that when replacing an asset all the places where the asset is used is also being modified to have the new ID?

**Comment by jasonvarga** [comment]
Yes it's on purpose. We used to store IDs but it was a nightmare. For most cases, it's nice to be able to drop a file path into your data.

Using IDs made managing the majority use case much more difficult. But I agree with you that it would absolutely be better in other situations.

However, if you use the control panel to move assets, all the references will be updated for you. Make sure you use the control panel rather than just moving files around using your OS.

**Reply by kevinmeijer97** [reply]
Thanks for your reply 😄 

I did make sure to move around the assets through the control panel, as i understand it wouldn't have been possible to update the references if i did it from my OS.

What i tried was the following scenario:

- We have a page collection which contains a replicator with a set where we can upload an image using the image field.
- The image was uploaded in the root folder, so it would be `/test-image.png`
- We then navigated to the Assets page in the CP and moved the image to a new folder so the new path would be `/test/test-image.png`
- After moving the image the image field on the page was empty.
- When we move the image back to the root folder the image appears again on the page.

If i understand it correctly this should have worked, maybe it's also good to note that we are using the markdown entries for our content.
